src/dds_manager.cpp

```diff
@@ -1060,6 +1136,88 @@ bool DDSManager::replaceFilter(const std::string& topicName,
 
 } // End DDSManager::replaceFilter
 
+bool DDSManager::replaceFilterParams(const std::string& topicName,
+    const std::string& readerName,
+    const DDS::StringSeq filterParams)
+{
+
+    // Make sure the data reader name is valid
+    if (readerName.empty())
+    {
+        std::cerr << "Error replacing topic filter for '"
+            << topicName
+            << "'. The reader name must not be empty."
+            << std::endl;
+
+        return false;
+    }
+
+
+    decltype(m_sharedLock) lock(m_topicMutex);
+    // Has the subscriber already been registered?
+    DDS::Subscriber_var subscriber = getSubscriber(topicName);
+    if (!subscriber)
+    {
+        std::cerr << "Error replacing topic filter for '"
+            << topicName
+            << "'. The subscriber has not been created."
+            << std::endl;
+
+        return false;
+    }
+
+    // Make sure this data reader exists
+    DDS::DataReader_var dataReader = getReader(topicName, readerName);
+    if (!dataReader)
+    {
+        std::cerr << "Error replacing topic filter for '"
+            << topicName
+            << "'. The data reader named '"
+            << readerName
+            << "' does not exist."
+            << std::endl;
+
+        return false;
+    }
+
+    // Get the ContentFilteredTopic
+    std::shared_ptr<TopicGroup> topicGroup = m_topics[topicName];
+
+
+    DDS::ContentFilteredTopic* topicDesc =
+        dynamic_cast<DDS::ContentFilteredTopic*>
+        (dataReader->get_topicdescription());
+
+    bool status = false;
+
+    // Update this content filtered topic if it exists
+    if (topicDesc)
+    {
+        for (auto iter = topicGroup->filteredTopics.begin();
+            iter != topicGroup->filteredTopics.end();
+            ++iter)
+        {
+            if (topicDesc != iter->second)
+            {
+                continue;
+            }
+            else
+            {
+                if (iter->second->set_expression_parameters(filterParams) == DDS::RETCODE_OK)
+                {
+                    status = true;
+                    break;
+                }
+                else
+                {
+                    break;
+                }
```

Comment:
Both sides of the if/else end in `break;`, so the `break;` can be factored out.

Reply:
Done.